From dd5f191ff010998931c8a9ed5789f03bb60a143e Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@compasses.com>
Date: Mon, 9 Nov 2020 09:12:04 -0600
Subject: [PATCH 28/63] RSOM-186

arm64: dts: Beacon EmbededdedWorks: Make clocking more explicit
add clock [parent] to wm8962 codec
change audio_clk_a to fixed factor and assign versaclock as parent

Signed-off-by: Charles Stevens <charles.stevens@compasses.com>
---
 .../dts/renesas/beacon-renesom-baseboard.dtsi  | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
index cdaa9172ca69..aa6f1e51f8cf 100755
--- a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
+++ b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
@@ -227,7 +227,12 @@
 };
 
 &audio_clk_a {
-	clock-frequency = <24576000>; /* Support 48KHz */
+	/delete-property/ clock-frequency;
+	#clock-cells = <0>;
+	compatible = "fixed-factor-clock";
+	clock-mult = <1>;
+	clock-div = <1>;
+	clocks = <&versaclock6_bb 4>;
 };
 
 &audio_clk_b {
@@ -285,14 +290,14 @@
 &ehci0 {
 	dr_mode = "otg";
 	status = "okay";
-	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 3>, <&versaclock6_bb 4>;
+	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 3>;
 };
 
 &ehci1 {
 	status = "okay";
-	/* Hack, we are asking the EHCI to take a clk ref to the Ethernet Phy
-	refclock so Ethernet will work */
-	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 4>, <&versaclock6_bb 3>;
+	/* Hack, we are asking the EHCI to take a clk ref to the Ethernet
+	refclock so Ethernet will work (AVB_TXCREFCLK) */
+	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 3>, <&versaclock5 4>;
 };
 
 &hdmi0 {
@@ -396,6 +401,7 @@
 		clocks = <&x304_clk>;
 		clock-names = "xin";
 		/* CSI0_MCLK, CSI1_MCLK, AUDIO_CLK_CODEC, AUDIO_CLKA */
+		clock-output-names = "versaclock6_bb.out0_sel_i2cb", "versaclock6_bb.out1", "versaclock6_bb.out2", "versaclock6_bb.out3", "versaclock6_bb.out4";
 		assigned-clocks = <&versaclock6_bb 1>, <&versaclock6_bb 2>, <&versaclock6_bb 3>, <&versaclock6_bb 4>;
 		assigned-clock-rates =	<24000000>, <24000000>, <24000000>, <24576000>;
 
@@ -444,6 +450,8 @@
 	codec: wm8962@1a {
 		compatible = "wlf,wm8962";
 		reg = <0x1a>;
+		clocks = <&versaclock6_bb 3>;
+		clock-names = "mclk";
 		DCVDD-supply = <&reg_audio>;
 		DBVDD-supply = <&reg_audio>;
 		AVDD-supply = <&reg_audio>;
-- 
2.17.1

